src/mbgl/renderer/paint_parameters.cpp

```diff
@@ -103,53 +103,71 @@ gfx::DepthMode PaintParameters::depthModeFor3D() const {
     return gfx::DepthMode{gfx::DepthFunctionType::LessEqual, gfx::DepthMaskType::ReadWrite, {0.0, depthRangeSize}};
 }
 
-void PaintParameters::clearStencil() {
-    nextStencilID = 1;
-    tileClippingMaskIDs.clear();
-
-#if MLN_RENDER_BACKEND_METAL
-    // Metal doesn't have an equivalent of `glClear`, so we clear the buffer by drawing zero to (0:0,0)
-    std::set<UnwrappedTileID> ids{{0, 0, 0}};
-    auto f = [](const auto& ii) -> const UnwrappedTileID& {
-        return ii;
-    };
-    renderTileClippingMasks(ids.cbegin(), ids.cend(), std::move(f), /*clear=*/true);
-#else
-    context.clearStencilBuffer(0b00000000);
-#endif
-}
-
 namespace {
 
 template <typename TIter>
 using GetTileIDFunc = std::function<const UnwrappedTileID&(const typename TIter::value_type&)>;
 using TileMaskIDMap = std::map<UnwrappedTileID, int32_t>;
 
-// Detects a difference in keys of renderTiles and tileClippingMaskIDs
+// `std::includes` adapted to unwrap arbitrary collection iterators
+template <class I1, class I2>
+bool includes(I1 beg1, const I1 end1, I2 beg2, const I2 end2, const GetTileIDFunc<I2>& f) {
```

Comment:
It's not really arbitrary because of the `GetTileIDFunc` type. Maybe make that parameter a template (too) and move it to its own file?

Reply:
It works on any iterable container of tile IDs, but not arbitrary containers of anything, which is why it's there instead of in `util` somewhere.  It could be made more general, but the un-wrapper function would get more complex.  All of this can be deleted entirely when the legacy code is removed, it's just to avoid duplicating the code for iterating `RenderTiles`.